homeassistant/components/fastdotcom/__init__.py

```diff
@@ -48,21 +44,20 @@ async def async_setup_platform(hass: HomeAssistant, config: ConfigType) -> bool:
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up the Fast.com component."""
-    data = hass.data[DOMAIN] = SpeedtestData(hass)
+    """Set up Fast.com from a config entry."""
+    coordinator = FastdotcomDataUpdateCoordindator(hass)
 
-    entry.async_on_unload(
-        async_track_time_interval(hass, data.update, timedelta(hours=DEFAULT_INTERVAL))
-    )
-    # Run an initial update to get a starting state
-    await data.update()
-
-    async def update(service_call: ServiceCall | None = None) -> None:
-        """Service call to manually update the data."""
-        await data.update()
+    async def _request_refresh(event: Event) -> None:
+        """Request a refresh."""
+        await coordinator.async_request_refresh()
 
-    hass.services.async_register(DOMAIN, "speedtest", update)
+    if hass.state == CoreState.running:
```

Comment:
Use the `async_at_started` helper instead.

Reply:
Your feedback is to wrap the service registration, in the `async_at_started`?

Reply:
No. We have a helper to handle calling a function either immediately if Home Assistant is already running or at the Home Assistant started event. Remove all the boilerplate for that logic here, and just use the helper `async_at_started`.

Reply:
I see I misread, your focus is on the CoreState.running, not on how to register a service.

Reply:
Yes. The first line above the comment is always the line that is marked when starting a single line comment in the GitHub UI.